webui/src/pages/repositories/repository/settings/retention.jsx

```diff
@@ -0,0 +1,187 @@
+import React, {useState} from "react";
+import {RepositoryPageLayout} from "../../../../lib/components/repository/layout";
+import {
+    ActionGroup,
+    ActionsBar,
+    Error,
+    Loading,
+    RefreshButton,
+    ToggleSwitch
+} from "../../../../lib/components/controls";
+import Button from "react-bootstrap/Button";
+import {NotFoundError, retention} from "../../../../lib/api";
+import {useAPI} from "../../../../lib/hooks/api";
+import {RefContextProvider, useRefs} from "../../../../lib/hooks/repo";
+import Card from "react-bootstrap/Card";
+import Table from "react-bootstrap/Table";
+
+import {SettingsLayout} from "./layout";
+import {Container, Row} from "react-bootstrap";
+import Col from "react-bootstrap/Col";
+import {PolicyEditor} from "../../../../lib/components/policy";
+
+const exampleJson = (defaultBranch) => {
+    return {
+        "default_retention_days": 21,
+        "branches":
+            [
+                {"branch_id": defaultBranch, "retention_days": 28},
+            ]
+    }
+}
+
+const GetStarted = ({repo, onSubmit}) => {
+    const [showCreate, setShowCreate] = useState(false);
+    return (
```

Comment:
I don't like "retention duration in lakeFS for deleted objects" text since it is too vague. Suggestion for the text:

> This policy determines for how long objects are kept in the storage after they are deleted in lakeFS. For more information, see the docs.

As far as I'm concerned, I wouldn't mind if this text appears also when the policy is already filled, but it's up to you.